### PR TITLE
[release script] Check for //NORELEASE in code before release

### PR DESCRIPTION
**Pull request by brwe**
Lines in the code that should be removed before a release can be annotated with
//NORELEASE . This can be useful when debugging test failures. For example,
one might want to add additional logging that would be too verbose for production
and therfore should be removed before releasing.

### CODE REVIEW

**Comment by s1monw** [comment]
LGTM

**Comment by mikemccand** [comment]
The matching is case insensitive right?

**Comment by s1monw** [comment]
`re.IGNORECASE` <== @mikemccand

File: dev-tools/build_release.py [hunk 1]
```diff
@@ -583,6 +583,20 @@ def ensure_checkout_is_clean(branchName):
   if 'is ahead' in s:
     raise RuntimeError('git status shows local commits; try running "git fetch origin", "git checkout %s", "git reset --hard origin/%s": got:\n%s' % (branchName, branchName, s))
 
+# Checks all source files for //NORELEASE comments
+def check_norelease(path='src'):
+  pattern = re.compile(r'\bnorelease\b', re.IGNORECASE)
+  for root, _, file_names in os.walk(path):
+    for file_name in fnmatch.filter(file_names, '*.java'):
```
Comment (mikemccand) [inline dev-tools/build_release.py]: Maybe also check .asciidoc, .json, .xml?

**Comment by mikemccand** [comment]
> re.IGNORECASE <== @mikemccand

Super :)

**Comment by mikemccand** [comment]
LGTM

**Pull request closed without merging**